systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -1671,6 +1675,145 @@ void testRegenerateCertExternalAddressChange() throws InterruptedException {
         assertNotEquals(secretsWithoutExt.getData().get("my-cluster-kafka-2.key"), secretsWithExt.getData().get("my-cluster-kafka-2.key"));
     }
 
+    @Test
+    void testLabelModificationDoesNotBreakCluster() throws Exception {
+        Map<String, String> labels = new HashMap<>();
+        String[] labelKeys = {"label-name-1", "label-name-2"};
+        String[] labelValues = {"name-of-the-label-1", "name-of-the-label-2"};
+        String[] labelNewValues = {"new-name-of-the-label-1", "new-name-of-the-label-2"};
+        String brokerServiceName = "my-cluster-kafka-brokers";
+        String configMapName = "my-cluster-kafka-config";
+
+        labels.put(labelKeys[0], labelValues[0]);
+        labels.put(labelKeys[1], labelValues[1]);
+
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 3, 1)
+                .editSpec()
+                    .editKafka()
+                        .editListeners()
+                            .withNewKafkaListenerExternalNodePort()
+                            .endKafkaListenerExternalNodePort()
+                        .endListeners()
+                    .endKafka()
+                .endSpec()
+                .editMetadata()
+                    .withLabels(labels)
+                .endMetadata()
+                .done();
+
+        LOGGER.info("Waiting for kafka service labels changed {}", labels);
+        StUtils.waitForKafkaServiceLabelsChange(brokerServiceName, labels);
```

Comment:
You seem to wait for labels at Service resource. But two lines below you check the labels on StatefulSet. I would this again suspect to be sensitive to race conditions, since the Service and StatefulSet are created independently.